### PR TITLE
Option to toggle restart policy, automatically stop iis if running, upgrade to SMS 4.1.0

### PR DESCRIPTION
**Pull request by navancommits**
…

## Description
Option to toggle restart policy, automatically stop iis if running, upgrade to SMS 4.1.0

## Motivation
Provides flexibility to users and saves time chasing errors

## How Has This Been Tested?
The iis stop script in up.ps1 has been working fine in my local env for at least few months, in case of others, the blogs are here:
SMS 4.1.0: https://navansitecorenotes.blogspot.com/2022/01/how-easy-it-is-to-upgrade-to-sitecore.html
Docker restart policy: https://navansitecorenotes.blogspot.com/2022/01/docker-restart-policy-set-to-always.html

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x ] Documentation update (non-breaking change; modified files are limited to the `/docs` directory)

## Checklist:

- [x ] I have read the Contributing guide.
- [x ] My code follows the code style of this project.
- [ x] My code/comments/docs fully adhere to the Code of Conduct.
- [x ] My change is a code change and it requires an update to the documentation.
- [ ] My change is a documentation change and it requires an update to the navigation.

### CODE REVIEW

**Review by robearlam — CHANGES_REQUESTED** [review]
Can you please move all of the changes from docker-compose.yml, to the docker-compose.override.yml.

The idea here is that the docker-compose.yml is the default OOTB asset and we make all solution specific changes in the override to make future upgrades simpler.l

File: docker-compose.override.yml [hunk 1]
```diff
@@ -89,6 +95,7 @@ services:
   mssql-init:
     isolation: ${ISOLATION}
     image: ${MVP_DOCKER_REGISTRY}mvp-mssql-init:${SITECORE_VERSION}
+    restart: ${DOCKER_RESTART}
```
Comment (robearlam) [inline docker-compose.override.yml:98]: Do we want to restart the MSSQL Init container job? 

This should terminate once the job is complete.

File: docker-compose.override.yml [hunk 2]
```diff
@@ -97,19 +104,22 @@ services:
...
 
   solr:
+    restart: ${DOCKER_RESTART}
     volumes:            
       - ${LOCAL_DATA_PATH}\solr:c:\data  
     mem_limit: ${MEM_LIMIT_SOLR:-1GB}  
 
   solr-init:
     image: ${MVP_DOCKER_REGISTRY}mvp-solr-init:${SITECORE_VERSION}
+    restart: ${DOCKER_RESTART}
```
Comment (robearlam) [inline docker-compose.override.yml:114]: Do we want to restart the SOLR Init container job? 

This should terminate once the job is complete.

Comment (navancommits) [reply docker-compose.override.yml:114]: Sure, will check and update....

**Review by robearlam — APPROVED** [review]

**Pull request merged**